### PR TITLE
[Docs] Update liquid clustering documentation for Delta 3.2

### PR DESCRIPTION
**Pull request by zedtang**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [x] Other (Doc)

## Description

Update documentation about liquid clustering for Delta 3.2

## How was this patch tested?

127 0 0 1_8000_delta-clustering html (5)

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

File: docs/source/delta-clustering.md [hunk 1]
```diff
@@ -61,23 +61,53 @@ Use the `OPTIMIZE` command on your table, as in the following example:
 OPTIMIZE table_name;
 ```
 
+Liquid clustering is incremental, meaning that data is only rewritten as necessary to accommodate data that needs to be clustered. Data files with clustering columns that do not match data to be clustered are not rewritten.
```
Comment (dabao521) [inline docs/source/delta-clustering.md:64]: > Data files with clustering columns that do not match data to be clustered are not rewritten.

Is it better say `Already clustered data files with different clustering columns are not rewritten`?

Comment (dabao521) [reply docs/source/delta-clustering.md:64]: Also should we mention that we still cluster the zcubes that are not full ( i.e. not sealed zcubes)?

**Review by dabao521 — APPROVED** [review]
pending comments

File: docs/source/delta-clustering.md [hunk 2]
```diff
@@ -61,23 +61,53 @@ Use the `OPTIMIZE` command on your table, as in the following example:
...
 ## Limitations
 
 The following limitations exist:
 
-- In <Delta> 3.1, users needs to enable the feature flag `spark.databricks.delta.clusteredTable.enableClusteringTablePreview` to use liquid clustering. The following features are not supported in this preview:
+- You can only specify columns with statistics collected for clustering columns. By default, the first 32 columns in a Delta table have statistics collected.
+- You can specify up to 4 clustering columns.
+
+.. important::
+  In <Delta> 3.1, users needs to enable the feature flag `spark.databricks.delta.clusteredTable.enableClusteringTablePreview` to use liquid clustering. The following features are not supported in this preview:
```
[the remark this answers is not in the document]
Comment (zedtang) [reply docs/source/delta-clustering.md:108]: Added a sentence below to clarify this.

File: docs/source/delta-clustering.md (hunk 1 above)
Comment (zedtang) [reply docs/source/delta-clustering.md]: > > Data files with clustering columns that do not match data to be clustered are not rewritten.
> 
> Is it better say `Already clustered data files with different clustering columns are not rewritten`?

Updated.

Comment (zedtang) [reply docs/source/delta-clustering.md]: > Also should we mention that we still cluster the zcubes that are not full ( i.e. not sealed zcubes)?

I think the `data is only rewritten as necessary` serves this purpose. I didn't want to mention too many implementation details.

**Review by imback82 — APPROVED** [review]

**Pull request merged**